pkg/util/client.go

```diff
@@ -32,6 +32,11 @@ type DoRequestFunc func(request *http.Request) (*http.Response, error)
 
 // SendRequest sends a request to the specified client and the provided URL with the specified parameters and body.
 func SendRequest(ctx context.Context, doRequest DoRequestFunc, method, url string, params map[string]string, body interface{}) (*http.Response, error) {
+	return SendRequestWithHeaders(ctx, doRequest, method, url, params, body, map[string]string{})
+}
+
+// SendRequestWithHeaders sends a request to the specified client and the provided URL with the specified parameters, body and headers.
+func SendRequestWithHeaders(ctx context.Context, doRequest DoRequestFunc, method, url string, params map[string]string, body interface{}, headers map[string]string) (*http.Response, error) {
```

Comment:
too many arguments, a struct would be better and also allow for optional arguments